app/uk/gov/hmrc/nisp/models/NpsDate.scala

```diff
@@ -16,29 +16,30 @@
 
 package uk.gov.hmrc.nisp.models
 
-import org.joda.time.LocalDate
-import org.joda.time.format.DateTimeFormat
+import java.time.LocalDate
+import java.time.format.DateTimeFormatter
 import play.api.libs.json._
 import uk.gov.hmrc.nisp.utils.Constants
 
 case class NpsDate(localDate: LocalDate) {
-  val toNpsString: String = NpsDate.dateFormat.print(localDate)
+  val toNpsString: String = NpsDate.dateFormat.format(localDate)
 
   val taxYear: Int = {
-    val year = localDate.year.get
-    if (localDate.isBefore(new LocalDate(year, Constants.taxYearsStartEndMonth, Constants.taxYearStartDay))) year - 1
+    val year = localDate.getYear
+    if (localDate.isBefore(LocalDate.of(year, Constants.taxYearsStartEndMonth, Constants.taxYearStartDay))) year - 1
     else year
   }
 }
 
 object NpsDate {
-  private val dateFormat   = DateTimeFormat.forPattern("dd/MM/yyyy")
+  private val pattern = "dd/MM/yyyy"
+  private val dateFormat   = DateTimeFormatter.ofPattern(pattern)
   private val npsDateRegex = """^(\d\d)/(\d\d)/(\d\d\d\d)$""".r
 
   implicit val reads = new Reads[NpsDate] {
     override def reads(json: JsValue): JsResult[NpsDate] =
       json match {
-        case JsString(npsDateRegex(d, m, y)) => JsSuccess(NpsDate(new LocalDate(y.toInt, m.toInt, d.toInt)))
+        case JsString(npsDateRegex(d, m, y)) => JsSuccess(NpsDate(LocalDate.of(y.toInt, m.toInt, d.toInt)))
```

Comment:
By capturing the date in this way. You don't need the formatter on line 36. However, using regex here feels a bit clunky. You could keep the formatter and instead of use LocalDate.of. Use LocalDate.parse(str, formatter)